src/mono/wasm/runtime/jiterpreter-interp-entry.ts

```diff
@@ -454,6 +483,12 @@ function generate_wasm_body (
 ) : boolean {
     // FIXME: This is not thread-safe, but the alternative of alloca makes the trampoline
     //  more expensive
+    // The solution is likely to put the address of the scratch buffer in a global that we provide
+    //  at module instantiation time, so each thread can malloc its own copy of the buffer
+    //  and then pass it in when instantiating instead of compiling the constant into the module
+    // FIXME: Pre-allocate these buffers and their constant slots at the start before we
```

Comment:
it would have to be dealocated when the worker shuts down, otherwise it would leak memory long term.

Reply:
Is worker shutdown a common occurrence?

Reply:
@lambdageek could tell you more, but I think yes.